### PR TITLE
feat(ssh_config): host_key_algorithms option

### PR DESCRIPTION
**Pull request by koralowiec**
##### SUMMARY

`host_key_algorithms` option for ssh_config module

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

ssh_config

##### ADDITIONAL INFORMATION

Example ad hoc execution:

```bash
ansible localhost --connection=local -a \
  "host=example.com ssh_config_file=/tmp/ssh_config remote_user=mariusz host_key_algorithms=+ssh-rsa" \
  -m community.general.ssh_config
```

Result (content of `/tmp/ssh_config`):

```
Host example.com
    user mariusz
    hostkeyalgorithms +ssh-rsa
```

### CODE REVIEW

File: changelogs/fragments/5605-ssh-config-add-host-key-algorithms.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - ssh_config - add `host_key_algorithms` option
```
Comment (felixfontein) [inline changelogs/fragments/5605-ssh-config-add-host-key-algorithms.yaml:2]:
```suggestion
  - ssh_config - add ``host_key_algorithms`` option (https://github.com/ansible-collections/community.general/pull/5605).
```

File: plugins/modules/ssh_config.py [hunk 2]
```diff
@@ -88,6 +88,10 @@
       - If I(user) and this option are not specified, C(/etc/ssh/ssh_config) is used.
       - Mutually exclusive with I(user).
     type: path
+  host_key_algorithms:
+    description:
+      - Sets the C(HostKeyAlgorithms) option.
+    type: str
```
Comment (felixfontein) [inline plugins/modules/ssh_config.py:94]:
```suggestion
    type: str
    version_added: 6.1.0
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution. Here are some first comments.

**Comment by koralowiec** [comment]
Hi @felixfontein , I included your suggestions in the latest commit

**Review by felixfontein — APPROVED** [review]
Looks OK to me, and since nobody objected I'm going to merge this.

**Comment by felixfontein** [comment]
@koralowiec thanks for contributing this feature!

**Pull request merged**